database/mongodb/mongodb_test.go

```diff
@@ -104,3 +107,111 @@ func TestWithAuth(t *testing.T) {
 			}
 		})
 }
+
+func TestTransaction(t *testing.T) {
+	versionsSupportedTransactions := []mt.Version{
+		{
+			Image: "mongo:4",
+			Cmd: []string{
+				"/bin/bash",
+				"-c",
+				"mongod --fork --logpath /data/log.log --bind_ip_all --replSet rs0 && mongo --eval 'rs.initiate()' && tail -f /data/log.log",
+			}},
+	}
+	mt.ParallelTest(t, versionsSupportedTransactions, isReady,
+		func(t *testing.T, i mt.Instance) {
+			//We should wait for replica init
+			//atm driver can't determine during initialization primary node without reconnect
+			time.Sleep(5 * time.Second)
```

Comment:
I'm wary about this since the tests need to be reliable. We already have issues w/ various DB tests arbitrarily timing out and I don't want to add additional ones that we know will have issues.

Use events instead of waiting for an arbitrary amount of time.
Can you use `rs.status()`? https://docs.mongodb.com/manual/reference/method/rs.status/

Reply:
Changed test, now i'm using `rs.initiate()` via code, not bash cmd, and `rs.status()` for checking replica set initializtion